base/compiler/typeinfer.jl

```diff
@@ -672,9 +672,7 @@ function type_annotate!(sv::InferenceState, run_optimizer::Bool)
                 deleteat!(src.codelocs, i)
                 deleteat!(sv.stmt_info, i)
                 nexpr -= 1
-                if oldidx < length(changemap)
-                    changemap[oldidx + 1] = -1
```

Comment:
Wha? Was this a bug?

Reply:
No, the changemap was just offset by one, but as a result, nothing was recorded for the last statement, but here I do need to know if the last statement got deleted (so I can update the relevant phi nodes if any), so I changed the ordering to be non-offset.